cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -197,25 +325,102 @@ private File getLatestFolder(File baseDirectory) {
+  private void addToStatus(SupportBundleStatus bundleStatus, String basePath) throws IOException {
     try (FileWriter statusFile = new FileWriter(new File(basePath, SupportBundleFileNames.STATUS_FILE_NAME))) {
-      GSON.toJson(supportBundleStatus, statusFile);
+      GSON.toJson(bundleStatus, statusFile);
     }
   }
 
+  /**
+   * read status file
+   */
   private SupportBundleStatus readStatusJson(File statusFile) throws IOException {
-    SupportBundleStatus supportBundleStatus;
+    SupportBundleStatus bundleStatus;
     try (Reader reader = Files.newBufferedReader(statusFile.toPath(), StandardCharsets.UTF_8)) {
-      supportBundleStatus = GSON.fromJson(reader, SupportBundleStatus.class);
+      bundleStatus = GSON.fromJson(reader, SupportBundleStatus.class);
     }
-    return supportBundleStatus;
+    return bundleStatus;
   }
 
+  /**
+   * valid if the namespace exists or not
+   */
   private NamespaceId validNamespace(NamespaceId namespace) throws Exception {
     if (!namespaceQueryClient.exists(namespace)) {
       throw new NamespaceNotFoundException(namespace);
     }
 
     return namespace;
   }
+
+  /**
+   * Collect single support bundle task status
+   */
+  private SupportBundleTaskReportStatus collectSupportBundleTaskStatus(
+    Set<SupportBundleTaskStatus> supportBundleTaskStatusSet) {
+    CollectionState systemLogTaskStatus = CollectionState.INVALID;
+    CollectionState pipelineInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeLogTaskStatus = CollectionState.INVALID;
+    CollectionState vmInfoTaskStatus = CollectionState.INVALID;
+    // Expand all supportBundleTaskStatusSet and map each individual taskType and subTaskType with corresponding status
+    // SupportBundleRuntimeInfoTask & SupportBundlePipelineRunLogTask are two subtasks of SupportBundlePipelineInfoTask
+    Map<SupportBundleTaskType, CollectionState> taskStatusMap = new HashMap<>();
+    for (SupportBundleTaskStatus supportBundleTaskStatus : supportBundleTaskStatusSet) {
+      SupportBundleTaskType currentTaskType = SupportBundleTaskType.valueOf(supportBundleTaskStatus.getType());
+      taskStatusMap.put(currentTaskType, supportBundleTaskStatus.getStatus());
+      if (supportBundleTaskStatus.getSubTasks().size() > 0) {
+        for (SupportBundleTaskStatus subTaskSupportBundleTaskStatus : supportBundleTaskStatus.getSubTasks()) {
+          SupportBundleTaskType currentSubTaskType =
+            SupportBundleTaskType.valueOf(subTaskSupportBundleTaskStatus.getType());
+          taskStatusMap.put(currentSubTaskType, subTaskSupportBundleTaskStatus.getStatus());
+        }
+      }
+    }
+
+    for (SupportBundleTaskType taskType : taskStatusMap.keySet()) {
+      switch (taskType) {
+        case SupportBundleSystemLogTask:
+          systemLogTaskStatus = taskStatusMap.get(taskType);
```

Comment:
What's the value of this enum? I mean we just want to return status of all tasks, why do we need to recategorize them? Can we use just a `List<SupportBundleTaskStatus>`  with each entry having  task name, status, potentially something else.

Reply:
I will probably drop this part for now, we need to revisit the format